ibis/backends/duckdb/__init__.py

```diff
@@ -134,6 +134,7 @@ def do_connect(
                 )
 
         if temp_directory is not None:
+            Path(temp_directory).mkdir(parents=True, exist_ok=True)
```

Comment:
Does duckdb clean up these temporary directories? If not, should we (with `atexit` or something)?

Reply:
Ah, it looks like duckdb does clean these up, yay!